asmcomp/linscan.ml

```diff
@@ -65,50 +65,31 @@ let release_expired_spilled ci pos =
       expired ci.ci_free_slots;
   ci.ci_spilled <- rest
 
+let dummy_interval pos =
+  {Interval.reg = Reg.dummy;
+   ibegin = pos;
+   iend = pos;
+   ranges = []}
+
 let release_expired_fixed ci pos =
-  let rec loop ci pos fixed il =
-    match IntervalSet.min_elt_opt il with
-    | Some i when i.iend >= pos ->
-        Interval.remove_expired_ranges i pos;
-        loop ci pos (IntervalSet.add i fixed) (IntervalSet.remove i il)
-    | _ ->
-        ci.ci_fixed <- fixed
-  in
-  loop ci pos IntervalSet.empty ci.ci_fixed
+  let (rest, divider_in_set, _expired) = IntervalSet.split (dummy_interval pos) ci.ci_fixed in
```

Comment:
I think for this to be correct, you would need to ensure that `dummy_interval pos` is greater than all intervals with `i.iend >= pos` and smaller than all intervals with `i.iend < pos`, which would mean inverting the direction of comparison of the stamps in the definition of `IntervalSet`, and also modifying `Reg.dummy` to have stamp `-1`, as currently it is possible to get another register with stamp `0`.

Reply:
Remark: the `-1` stamp comes in a later fix, and the inversion of the comparison is in my commit https://github.com/nojb/ocaml/pull/11/commits/d78e8f9e6016c8d900af384502703ee08fce9083 in https://github.com/nojb/ocaml/pull/11